dasharo-security/sort-pci-semiauto.robot

```diff
@@ -0,0 +1,79 @@
+*** Settings ***
+Library             Collections
+Library             DateTime
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Variables ***
+${LSPCI_CMD}=       lspci -vv
+${IP_LINK_CMD}=     ip link
+
+
+*** Test Cases ***
+Sort000.000 Port Order and PCIe Switching
+    [Documentation]    This test automates the verification of port order based on PCIe bus numbers and checks PCIe switching.
+
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${pci_devices}=    Get PCIe Bus Info With MACs
+    Log    PCIe devices and their MACs: ${pci_devices}
+    Compare Interface And PCIe Bus Order    ${pci_devices}
+
+
+*** Keywords ***
+Get PCIe Bus Info With MACs
+    [Documentation]    Extract PCIe bus numbers and MAC addresses from lspci output.
+    ${lspci_output}=    Execute Command In Terminal    ${LSPCI_CMD}
+    ${lines}=    Split String    ${lspci_output}    \n
+    ${pci_devices}=    Create Dictionary
+    ${current_device}=    Set Variable    None
+    FOR    ${line}    IN    @{lines}
+        ${is_device_line}=    Evaluate    'Ethernet controller' in '${line}'
+        IF    ${is_device_line}
+            ${current_device}=    Get Device Bus Info    ${line}
+        END
+        ${is_mac_line}=    Evaluate    'Device Serial Number' in '${line}'
+        IF    ${is_mac_line}
+            Append PCIe MAC    ${line}    ${pci_devices}    ${current_device}
```

Comment:
We don't actually need to store the exact bus number. All we need is the ascending order of the devices/MACs sorted by the bus number. As we parse the lspci output line by line, we will get the ascending order already.

Reply:
the test checks the MACs and config list: https://github.com/Dasharo/open-source-firmware-validation/pull/515/commits/306e90f5e3d77337b3488d4b1832240b4b97bb86 is it enough?

Reply:
We don't store bus numbers now, however the keyword names still have the word bus  in their names. Will open new comments for that.